models/ROMS_rutgers/model_mod.f90

```diff
@@ -0,0 +1,1716 @@
+   call nc_check(nf90_put_var(ncid, varid, run_duration), routine, 'put_var')
+   return
+endif
+
+end subroutine write_model_time
+
+
+!--------------------------------------------------------------------
+! Read the time from the input file
+! filename: Name of file that contains the time
+
+function read_model_time(filename)
+
+character(len=*), parameter :: routine = 'read_model_time'
+
+character(len=*), intent(in) :: filename
+type(time_type)              :: read_model_time
+
+integer :: ncid
+
+if (.not. module_initialized) call static_init_model
+
+if (.not. file_exist(filename)) then
+   write(string1,*) 'cannot open file ', trim(filename),' for reading.'
+   call error_handler(E_ERR, routine, string1, source)
+endif
+
+ncid = nc_open_file_readonly(trim(filename), routine)
+
+call get_time_information(filename, ncid, 'ocean_time', 'ocean_time', last_time=read_model_time)
+call nc_close_file(ncid, routine)
+
+end function read_model_time
+
+
+!-----------------------------------------------------------------------
+! Set the desired minimum model advance time. This is generally NOT the
+! dynamical timestep of the model, but rather the shortest forecast length
+! you are willing to make. This impacts how frequently the observations
+! may be assimilated.
+
+function set_model_time_step()
+
+type(time_type) :: set_model_time_step
+
+! assimilation_period_seconds, assimilation_period_days are from the namelist
+set_model_time_step = set_time(assimilation_period_seconds, assimilation_period_days)
+
+end function set_model_time_step
+
+
+!-----------------------------------------------------------------------
+! Read the grid dimensions from the ROMS grid netcdf file.
+! By reading the dimensions first, we can use them in variable
+! declarations later - which is faster than using allocatable arrays.
+
+subroutine get_grid_dimensions()
+
+character(len=*), parameter :: routine = 'get_grid_dimensions'
+
+integer :: ncid
+integer :: Nxi_rho, Nxi_u, Nxi_v
+integer :: Neta_rho, Neta_u, Neta_v
+integer :: Ns_rho, Ns_w
+
+! Read the (static) grid dimensions from the ROMS grid file.
+ncid = nc_open_file_readonly(roms_filename, routine)
+
+Nxi_rho  = nc_get_dimension_size(ncid, 'xi_rho',  routine)
+Nxi_u    = nc_get_dimension_size(ncid, 'xi_u',    routine)
+Nxi_v    = nc_get_dimension_size(ncid, 'xi_v',    routine)
+Neta_rho = nc_get_dimension_size(ncid, 'eta_rho', routine)
+Neta_u   = nc_get_dimension_size(ncid, 'eta_u',   routine)
+Neta_v   = nc_get_dimension_size(ncid, 'eta_v',   routine)
+
+! Read the vertical dimensions 
+Ns_rho   = nc_get_dimension_size(ncid, 's_rho', routine)
+Ns_w     = nc_get_dimension_size(ncid, 's_w'  , routine)
+
+call nc_close_file(ncid, routine)
+
+Nx = Nxi_rho  ! Setting the nominal value of the 'global' variables
+Ny = Neta_rho ! Setting the nominal value of the 'global' variables
+Nz = Ns_rho   ! Setting the nominal value of the 'global' variables
```

Comment:
Are the grids always the same size for u,t,v? 
If they are this fine.

Reply:
yes!